sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -432,7 +433,7 @@ public ServiceBusSenderAsyncClient buildAsyncClient() {
                 case QUEUE:
                     entityName = queueName;
                     break;
-                case TOPIC:
+                case SUBSCRIPTION:
                     entityName = topicName;
```

Comment:
It is confusing to call it `SUBSCRIPTION` in sender. But I understand why you have done it.

Reply:
Yes. To match the entity type passed to Service Bus.